src/app/brukerprofil/kontaktinformasjon/KontaktinformasjonForm.tsx

```diff
@@ -0,0 +1,195 @@
+import * as React from 'react';
+import { ChangeEvent, FormEvent } from 'react';
+import { Action } from 'history';
+import { connect, Dispatch } from 'react-redux';
+import styled from 'styled-components';
+
+import KnappBase from 'nav-frontend-knapper';
+import UndertekstBold from 'nav-frontend-typografi/lib/undertekst-bold';
+import { Person } from '../../../models/person/person';
+import { AppState } from '../../../redux/reducer';
+import { KodeverkResponse } from '../../../models/kodeverk';
+import { Telefon } from '../../../models/person/NAVKontaktinformasjon';
+import { formaterHustelefonnummer, formaterMobiltelefonnummer } from '../../../utils/telefon-utils';
+import { TelefonInput, TelefonMetadata } from './TelefonInput';
+
+export interface TelefonInput {
+    retningsnummer: string;
+    telefonnummer: string;
+}
+
+interface State {
+    mobilInput: TelefonInput;
+    jobbTelefonInput: TelefonInput;
+    hjemTelefonInput: TelefonInput;
+}
+
+interface DispatchProps {
+}
+
+interface StateProps {
+}
+
+interface OwnProps {
+    person: Person;
+    retningsnummerKodeverk: KodeverkResponse;
+}
+
+type Props = DispatchProps & StateProps & OwnProps;
+
+const InputWrapper = styled.div`
+  margin-top: 1em;
+`;
+
+const TelefonWrapper = styled.div`
+  margin-bottom: 1em;
+`;
+
+function getInitialTelefonState(telefon: Telefon | undefined) {
+    return {
+        retningsnummer: telefon ? telefon.retningsnummer : '',
+        telefonnummer: telefon ? formaterMobiltelefonnummer(telefon.telefonnummer) : ''
+    };
+}
+
+class KontaktinformasjonForm extends React.Component<Props, State> {
+
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            mobilInput: getInitialTelefonState(this.props.person.kontaktinformasjon.mobil),
+            jobbTelefonInput: getInitialTelefonState(this.props.person.kontaktinformasjon.jobbTelefon),
+            hjemTelefonInput: getInitialTelefonState(this.props.person.kontaktinformasjon.hjemTelefon),
+        };
+
+        this.mobilTelefonnummerInputChange = this.mobilTelefonnummerInputChange.bind(this);
+        this.mobilRetningsnummerInputChange = this.mobilRetningsnummerInputChange.bind(this);
+        this.jobbTelefonnummerInputChange = this.jobbTelefonnummerInputChange.bind(this);
+        this.jobbRetningsnummerInputChange = this.jobbRetningsnummerInputChange.bind(this);
+        this.hjemTelefonnummerInputChange = this.hjemTelefonnummerInputChange.bind(this);
+        this.hjemRetningsnummerInputChange = this.hjemRetningsnummerInputChange.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    mobilTelefonnummerInputChange(event: ChangeEvent<HTMLInputElement>) {
+        this.setState({
+            mobilInput: {
+                telefonnummer: formaterMobiltelefonnummer(event.target.value),
+                retningsnummer: this.state.mobilInput.retningsnummer
```

Comment:
her settes vel en variabel til seg selv..?

Reply:
Stemmer, men for å sette den ene så må begge assignes.

Mulig det er mulig å kun sette den ene med noe javascript magi, men jeg fant ikke noe i farta.

Reply:
aha!
ja, da tror jeg den normale notasjonen er  noe slikt:
``````

Reply:
Se der ja 💪 💯 💹